assets/app/app.rb

```diff
@@ -58,15 +58,23 @@ def render_content
       when /[^?](game|hotseat|tutorial)/
         render_game
       when /signup/
-        h(View::User, user: @user, type: :signup)
+        return h('h3', 'You are already logged in') if @user
+
+        h(View::User, type: :signup)
```

Comment:
why don't you just do this in the user view?

Reply:
Surprisingly, there is a reason. In the view there is a similar `case` distinction, but it goes like this:
```
title, inputs =
  case @type
  when :signup
    ...      
```

And that doesn't allow to `return`

As I see it, there were 3 options:
1. The one I did, moving it "outside"
2. Repeating the lhs of the assignment in the 3 options
3. Add a third `case` in the beginning just to deal with this

If you see a 4th or prefer one of the other 2, I'll change it.

Reply:
what do you mean it doesn't allow return?

Reply:
I mean
`MiniRacer::SnapshotError: TypeError: value.$respond_to? is not a function`

I can't figure out why this is happening, but it only happens when I `return` somewhere inside the `title, inputs = ...`

Reply:
are you returning an h element?

Reply:
Yes

Reply:
hit me up on slack sometime and i can help you debug